### PR TITLE
log socket ID when changing to connected state

### PR DESCRIPTION
**Pull request by jameshfisher**

### CODE REVIEW

**Comment by jameshfisher** [comment]
tested manually, produces logs like

```
test_log_socketid.htm:9 Pusher : State changed : initialized -> connecting
test_log_socketid.htm:9 Pusher : Connecting : {"transport":"ws","url":"wss://ws.pusherapp.com:443/app/c95c18935a9f763327b1?protocol=7&client=js&version=3.0.0&flash=false"}
test_log_socketid.htm:9 Pusher : State changed : connecting -> connected with new socket ID 194496.582858
test_log_socketid.htm:9 Pusher : Event sent : {"event":"pusher:subscribe","data":{"channel":"test_channel"}}
test_log_socketid.htm:9 Pusher : Event recd : {"event":"pusher_internal:subscription_succeeded","data":{},"channel":"test_channel"}
test_log_socketid.htm:9 Pusher : No callbacks on test_channel for pusher:subscription_succeeded
```

File: src/connection/connection_manager.js [hunk 1]
```diff
@@ -340,7 +340,11 @@
     var previousState = this.state;
     this.state = newState;
     if (previousState !== newState) {
-      Pusher.debug('State changed', previousState + ' -> ' + newState);
+      var newStateDescription = newState;
+      if (newState === "connected") {
+        newStateDescription += " with new socket ID " + data.socket_id;
+      }
+      Pusher.debug('State changed', previousState + ' -> ' + newStateDescription);
```
Comment (topliceanu) [inline src/connection/connection_manager.js]: great!

**Comment by jameshfisher** [comment]
Anyone opposed to merging?

**Comment by vivangkumar** [comment]
Looks good, IMO.

**Pull request merged**